hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -993,6 +1040,12 @@ public final boolean equals(Object o) {
         if (quorumName != null ? !quorumName.equals(that.quorumName) : that.quorumName != null) {
             return false;
         }
+        if (merkleTreeConfig != null ? !merkleTreeConfig.equals(that.merkleTreeConfig) : that.merkleTreeConfig != null) {
```

Comment:
minor: if `merkleTreeConfig` and `eventJournalConfig` cannot be `null`, some null checks can be ommited for readability.